File: Chapter 4/XmlToParquet.scala
import org.apache.spark.sql.{SparkSession, SaveMode}

object XmlToParquet {
  def main(args: Array[String]): Unit = {
    if (args.length < 3) {
      throw new IllegalArgumentException("Usage: <input-xml-path> <output-parquet-path> <root-tag>")
    }

    val spark = SparkSession.builder()
      .appName("XmlToParquet")
      .getOrCreate()

    val xmlDataFrame = spark.read
      .format("com.databricks.spark.xml")
      .option("rowTag", args(2))
      .load(args(0))

    xmlDataFrame.write
      .mode(SaveMode.Overwrite)
      .parquet(args(1))

    spark.stop()
  }
}

File: Chapter2_running_hadoop_workloads/xml_to_parquet.py
import sys
from pyspark.sql import SparkSession
from pyspark.sql.functions import *
from pyspark.sql.types import *


spark = SparkSession.builder \
    .appName("XmltoParquet") \
    .master("yarn") \
    .getOrCreate()

#Get the input path, output path and rowTag
inputPath = sys.argv[1]
outputPath = sys.argv[2]
rowTag = sys.argv[3]

print("Reading the XML input file")

df = spark.read.format('xml').option("rowTag", rowTag).load(inputPath)

# Show the DataFrame to check the data
df.write.mode("overwrite").parquet(outputPath)

print("Application Completed!!!")

# Closing the Spark session
spark.stop()